whois-client/src/main/java/net/ripe/db/whois/common/rpsl/attrs/DsRdata.java

```diff
@@ -74,6 +91,18 @@ public static DsRdata parse(final String value) {
             throw new AttributeParseException("Invalid digest type: " + digestType, value);
         }
 
+        try {
+            if (!DsRdataDigestType.validateLength(digestType, digestAsHex)) {
+                throw new AttributeParseException("Digest format is invalid for digest type " + digestType + ": ", digestAsHex);
+            }
+        } catch (AttributeParseException e) {
+            // catch and throw previous AttributeParseException otherwise we'd catch it as an IllegalArgumentException below
+            throw e;
+        } catch (IllegalArgumentException e) {
+            // unknown type, let it through (could be a new type we do not know about yet)
+            return new DsRdata(keytag, algorithm, digestType, digestAsHex);
+        }
```

Comment:
Why catching an exception and throwing it directly again? I would earlier argue for a check of 'unknown' types and put a check on the known ones.